sourcegraph/sourcegraph
# cody feature: add button to copy code

RE[ feature requested by user](https://sourcegraph.slack.com/archives/C04MSD3DP5L/p1679516277669339)
Close https://github.com/sourcegraph/cody/issues/97

### 1. Add a button to copy code block text to each code block:

![image](https://user-images.githubusercontent.com/68532117/228306555-f63526fb-1161-4959-ae7a-d25a39893dc2.png)


### Fix HTML tags rendering issue in chat view

RE: https://github.com/sourcegraph/cody/issues/97

<table>
<tr>
<td>Before</td>
<td>After</td>
</tr>
<tr>
<td>
<img width="400" alt="image" src="https://user-images.githubusercontent.com/68532117/228063621-ff48d77a-685c-4cff-b0cf-f88cc9a7efe8.png">
<img width="400" alt="image" src="https://user-images.githubusercontent.com/68532117/228063940-0898805a-718e-4430-ae6e-ec04cfaa0860.png">
</td>
<td>
<img width="400" alt="image" src="https://user-images.githubusercontent.com/68532117/228063194-c09e8b38-0a07-4220-9ea7-640c15a5554d.png">
<img width="400" alt="image" src="https://user-images.githubusercontent.com/68532117/228063847-020c6370-2997-4450-a5a6-5a714d402341.png">
</td>
</tr>
</table>


## Test plan



To test:
1. clone and pull the latest commit from this branch
2. run pnpm install from root
3. Select Launch Cody Extension from the dropdown menu in the RUN AND DEBUG sidebar in VS Code
4. Type `what does <button>button</button> do in html?` to see HTML render issues resolved
5. Type `show me the code to create a button in react` to test copy button in code snippets. 

Demo:

![cody-copy-btn](https://user-images.githubusercontent.com/68532117/228307211-2f4fd6ce-ac19-4326-847c-8220d6c91050.gif)

## App preview:

- [Web](https://sg-web-bee-cody-copy.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
